tutorials/Inference-PyTorch-Bert-Model-for-High-Performance-in-ONNX-Runtime.ipynb

```diff
@@ -0,0 +1,344 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 0. Prerequisites ##\n",
+    "First you need to check if the following packages exist and install them if needed. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Install a pip package in the current Jupyter kernel\n",
+    "import sys\n",
+    "!{sys.executable} -m pip install wget\n",
+    "!{sys.executable} -m pip install tensorflow_datasets\n",
+    "!{sys.executable} -m pip install --user torch==1.3.1 torchvision==0.4.2+cpu -f https://download.pytorch.org/whl/torch_stable.html\n",
+    "!{sys.executable} -m pip install transformers"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Load Pretrained Bert Model from Hugging Face ##"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We begin by downloading the model files and store them in the specified location. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "# Create a directory to store predict file\n",
+    "output_dir = \"./pytorch_output\"\n",
+    "cache_dir = \"./pytorch_squad\"\n",
+    "predict_file = os.path.join(cache_dir, \"dev-v1.1.json\")\n",
+    "# create cache dir\n",
+    "if not os.path.exists(cache_dir):\n",
+    "    os.makedirs(cache_dir)\n",
+    "    \n",
+    "# Download the file\n",
+    "predict_file_url = \"https://rajpurkar.github.io/SQuAD-explorer/dataset/dev-v1.1.json\"\n",
+    "if not os.path.exists(predict_file):\n",
+    "    import wget\n",
+    "    print(\"Start downloading predict file.\")\n",
+    "    wget.download(predict_file_url, predict_file)\n",
+    "    print(\"Predict file downloaded.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Specify some model config variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define some variables\n",
+    "model_type = \"bert\"\n",
+    "model_name_or_path = \"bert-base-cased\"\n",
+    "max_seq_length = 128\n",
+    "doc_stride = 128\n",
+    "max_query_length = 64\n",
+    "per_gpu_eval_batch_size = 1\n",
+    "eval_batch_size = 1\n",
+    "import torch\n",
+    "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Start to load model from pretrained. This step could take a few minutes. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import (WEIGHTS_NAME, BertConfig, BertForQuestionAnswering, BertTokenizer)\n",
```

Comment:
is all this code specific to ONNX? If not, where did we get it? Can we reference a HuggingFace document that has it?